### PR TITLE
Display a matrix for Node2D and don't display a duplicate origin

### PR DESCRIPTION
**Pull request by aaronfranke**
This is a subset of #32068 because it's taking a long time to be merged and some have asked for a smaller PR.

This PR only changes the code for Node2D and Node3D, without changing the inspector code. For 2D, the matrix is displayed, while previously it just wasn't displayed at all. For 3D, "Matrix" is now "Raw Matrix", and it displays the Basis instead of Transform.

Before/after pic (left is before, right is after):

Untitled

### CODE REVIEW

**Comment by QbieShay** [comment]
If it displays only the basis, it should be called "raw basis" imo

**Comment by KoBeWi** [comment]
Isn't the last row of 2D matrix also a duplicate origin?

**Comment by aaronfranke** [comment]
@KoBeWi Yes, but there is no way to hide it without modifying the inspector code (what the other PR does), because there is no Basis2D type.

File: scene/3d/node_3d.cpp
Comment (KoBeWi) [inline scene/3d/node_3d.cpp:713]: Why the argument is called `local` instead of `basis`?

**Review by KoBeWi — CHANGES_REQUESTED** [review]
The new `basis` property in Node3D needs doc.

File: scene/3d/node_3d.cpp
Comment (aaronfranke) [reply scene/3d/node_3d.cpp:713]: The existing argument for `set_transform` is also called local. I can change this if desired though.

**Comment by aaronfranke** [comment]
Good catch, I updated the documentation.

File: scene/3d/node_3d.cpp
Comment (KoBeWi) [reply scene/3d/node_3d.cpp:713]: I think changing the argument name should be fine. There is already `set_global_transform` method, so it won't be confusing or something.

Comment (KoBeWi) [reply scene/3d/node_3d.cpp]: I actually meant that the argument name in `set_transform()` could be changed too (to `transform`).

**Review by KoBeWi — APPROVED** [review]
Looks ok.

We should look into removing the last row of 2D raw matrix, but it could be done in another PR.

**Review by fire — APPROVED** [review]
Looks good after review.

**Comment by fire** [comment]
Raw matrix implies showing 3x4 or 4x4 matrix for the transform. FYI.

**Review by mhilbrunner — APPROVED** [review]

**Comment by reduz** [comment]
@aaronfranke the fact you get an approval does not mean you can merge, specially when other contributors are hesitant about this. Please do not merge these type of PRs without consensus including mine.

**Comment by aaronfranke** [comment]
@reduz I thought that I did have consensus, and there has been lots of discussion recently about how maintainers have full control of the merge button. I don't understand what you don't like about this PR either. Keep in mind that this PR is *not* #32068.

**Comment by reduz** [comment]
Maintainers can only merge PRs for areas they maintain, or with permission of the person maintaining this area. You did not have that permission to merge this.

**Pull request merged**
